types/payload.go

```diff
@@ -121,16 +142,29 @@ func (p *Payload) Exists(key string) bool {
 		}
 	}
 
+	if p.missingFields == nil {
+		p.missingFields = make(map[string]struct{})
+	}
+	p.missingFields[key] = struct{}{}
```

Comment:
again, I don't think we should memoize by default

Reply:
Oops, sorry I missed this one